src/main/java/uk/gov/hmcts/reform/professionalapi/controller/external/OrganisationExternalController.java

```diff
@@ -425,4 +431,84 @@ public void deletePaymentAccountsOfOrganisation(
 
     }
 
+    @ApiOperation(
+            value = "Adds contact informations(address details) to organisation",
+            notes = "**IDAM Roles to access API** :\n pui-organisation-manager",
+            authorizations = {
+                    @Authorization(value = "ServiceAuthorization"),
+                    @Authorization(value = "Authorization")
+            }
+
+    )
+    @ApiResponses({
+            @ApiResponse(
+                    code = 201,
+                    message = ""
+            ),
+            @ApiResponse(
+                    code = 400,
+                    message = "An invalid request has been provided"
+            ),
+            @ApiResponse(
+                    code = 401,
+                    message = "Unauthorized Error : The requested resource is restricted and requires authentication"
+            ),
+
+            @ApiResponse(
+                    code = 403,
+                    message = "Forbidden Error: Access denied"
+            ),
+            @ApiResponse(
+                    code = 500,
+                    message = "Internal Server Error"
+            )
+    })
+
+    @PostMapping(
+            path = "/addresses",
+            consumes = APPLICATION_JSON_VALUE,
+            produces = APPLICATION_JSON_VALUE
+    )
+    @ResponseStatus(value = HttpStatus.CREATED)
+    @ResponseBody
+    @Secured({"pui-organisation-manager"})
+    public ResponseEntity<Void> addContactInformationsToOrganisation(
+            @NotNull @RequestBody List<@Valid ContactInformationCreationRequest> contactInformationCreationRequests,
+            @ApiParam(hidden = true) @OrgId String organisationIdentifier) {
+
+        ContactInformationEntityResponse contactInformationsResponse = null;
+
+        List<ContactInformationValidationResponse> contactInfoValidations =
+                organisationCreationRequestValidator.validate(contactInformationCreationRequests);
+        List<ContactInformationValidationResponse> result = null;
+        if (contactInfoValidations != null && !contactInfoValidations.isEmpty()) {
+            result = contactInfoValidations.stream()
+                    .filter(contactInfoValidation -> !contactInfoValidation.isValidAddress())
+                            .collect(Collectors.toList());
+        }
+        if (result != null && !result.isEmpty()) {
+            contactInformationsResponse = new ContactInformationEntityResponse();
+            contactInformationsResponse.setContactInfoValidations(contactInfoValidations);
+            throw new InvalidContactInformations("Invalid Contact informations", contactInfoValidations);
+        } else {
+
+            Optional<Organisation> organisation = Optional.ofNullable(organisationService
+                    .getOrganisationByOrgIdentifier(organisationIdentifier));
+
+            if (organisation.isEmpty()) {
+                throw new ResourceNotFoundException("Organisation does not exist");
+            }
```

Comment:
Can this code be moved to an applicable validator?

Reply:
resolved